src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1956 @@
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPastAndCrossAttentions]:
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        use_cache = use_cache if use_cache is not None else self.config.use_cache
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        # decoder outputs consists of (dec_features, past_key_value, dec_hidden, dec_attn)
+        decoder_outputs = self.decoder(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            past_key_values=past_key_values,
+            inputs_embeds=inputs_embeds,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        if not return_dict:
+            return decoder_outputs
+
+        return BaseModelOutputWithPastAndCrossAttentions(
+            last_hidden_state=decoder_outputs.last_hidden_state,
+            past_key_values=decoder_outputs.past_key_values,
+            hidden_states=decoder_outputs.hidden_states,
+            attentions=decoder_outputs.attentions,
+            cross_attentions=decoder_outputs.cross_attentions,
+        )
+
+
+@add_start_docstrings(
+    "The Clvp decoder model with a language modelling head on top.",
+    CLVP_START_DOCSTRING,
+)
+class ClvpForCausalLM(ClvpPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.config = config.decoder_config if hasattr(config, "decoder_config") else config
+        self.model = ClvpModel(self.config)
+
+        self.final_norm = nn.LayerNorm(self.config.hidden_size)
+        self.lm_head = nn.Linear(self.config.n_embd, self.config.vocab_size, bias=True)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.model.decoder.input_embeds_layer
+
+    def set_input_embeddings(self, new_embeddings):
+        self.model.decoder.input_embeds_layer = new_embeddings
+
+    def prepare_inputs_for_generation(
+        self, input_ids, past_key_values=None, inputs_embeds=None, conditioning_embeds=None, **kwargs
+    ):
+        # for the first pass, so we must add the start token to the conditioning_embeds and return them as `inputs_embeds`
+        if conditioning_embeds is not None and past_key_values is None:
+            # Add the start mel token at the end
+            mel_start_token_id = torch.tensor([[self.config.bos_token_id]], device=conditioning_embeds.device)
```

Comment:
This BOS pre-processing should be done in `.prepare_model_inputs` and not `prepare_inputs_for_generation`. See MusicGen for an example: https://github.com/huggingface/transformers/blob/1a2e966cfe916cdfd78ef7e1385911106bc33f50/src/transformers/models/musicgen/modeling_musicgen.py#L1290

Reply:
done